### PR TITLE
Datastore docs clean-up.

### PR DESCRIPTION
**Pull request by dhermes**
Somewhat inspired by #1069 (or at least the branch referenced there, which I recently deleted).

Also follows on #1345.

### CODE REVIEW

File: gcloud/datastore/client.py [hunk 1]
```diff
@@ -93,15 +93,14 @@ def _extended_lookup(connection, dataset_id, key_pbs,
...
+                    copied into it.
 
-    :type deferred: an empty list or None.
-    :param deferred: If a list is passed, the key protobufs returned
+    :type deferred: list
+    :param deferred: (Optional) If a list is passed, the key protobufs returned
                      by the backend as "deferred" will be copied into it.
-                     Use only as a keyword param.
 
     :type eventual: boolean
```
Comment (tseaver) [inline gcloud/datastore/client.py]: `bool`?

Comment (dhermes) [inline gcloud/datastore/client.py]: On it.

Comment (dhermes) [inline gcloud/datastore/client.py]: I figured I'd do a `git grep ': bool'` as a one-off later on.

File: gcloud/datastore/transaction.py [hunk 2]
```diff
@@ -60,15 +60,17 @@ class Transaction(Batch):
...
 
          >>> with datastore.Transaction():
          ...     entity = datastore.Entity(key=Key('Thing'))
          ...     datastore.put(entity)
-         ...     assert entity.key.is_partial  # There is no ID on this key.
+         ...     print entity.key.is_partial  # There is no ID on this key.
          ...
-         >>> assert not entity.key.is_partial  # There *is* an ID.
+         True
+         >>> entity.key.is_partial  # There *is* an ID.
```
Comment (tseaver) [inline gcloud/datastore/transaction.py]: Use `print` for consistency?

Comment (dhermes) [inline gcloud/datastore/transaction.py]: Works for me

**Comment by dhermes** [comment]
@tseaver Made the changes and rebased

**Comment by tseaver** [comment]
LGTM

Note that this is another case of squash-defeats-re-review.

**Comment by dhermes** [comment]
Ack on the squashing. Will change behavior (remind me if I forget). (Makes me miss Google's code review tool.)

**Comment by tseaver** [comment]
I've seen a Github feature request somewhere for a "Merge w/ squash" button.

**Comment by dhermes** [comment]
:+1: I like it

**Pull request merged**